contracts/contracts/GrantRound.sol

```diff
@@ -0,0 +1,134 @@
+pragma solidity ^0.8.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./GrantRegistry.sol";
+
+contract GrantRound {
+  using SafeERC20 for IERC20;
+  // @notice Unix timestamp of the start of the round
+  uint256 public startTime;
+  // @notice Unix timestamp of the end of the round
+  uint256 public endTime;
+  // @notice Contract owner
+  address public owner;
+  // @notice GrantsRegistry
+  address public registry;
+  // @notice ERC20 token that accepts pool donations
+  IERC20 public donationToken;
+  // @notice URL pointing to grant round metadata (for off-chain use)
+  string public metaPtr;
+  // @notice minimum contribution amt that can be made
+  uint256 public minContribution;
+  // @notice Set to true if grant round has ended and payouts have been released
+  bool public hasPaidOut;
+
+  /// @notice Emitted when a grant receives a donation
+  event DonationSent(uint96 indexed id, address indexed token, uint256 amount, address dest, address indexed donor);
+
+  modifier onlyOwner() {
+    require(msg.sender == owner, "Only owner can call this method");
+    _;
+  }
+
+  modifier beforeRoundEnd() {
+    require(block.timestamp < endTime, "Time has passed to complete this tx");
+    _;
+  }
+
+  modifier activeRound() {
+    require(block.timestamp >= startTime && block.timestamp < endTime, "Donations must be sent during an active round");
+    _;
+  }
+
+  modifier roundEnd() {
+    require(block.timestamp >= endTime, "Method must be called after the active round has ended");
+    _;
+  }
+
+  /**
+   * @notice Instantiates a new grant round
+   * @param _owner Grant round owner that has permission to payout the matching pool
+   * @param _registry Address that contains the grant metadata
+   * @param _donationToken Address of the ERC20 token in which donations are made
+   * @param _startTime Unix timestamp of the start of the round
+   * @param _endTime Unix timestamp of the end of the round
+   * @param _metaPtr URL pointing to the grant round metadata
+   * @param _minContribution Miniumum donation amount that can be made using the given donation token
+   */
+  constructor(
+    address _owner,
+    address _registry,
+    IERC20 _donationToken,
+    uint256 _startTime,
+    uint256 _endTime,
+    string memory _metaPtr,
+    uint256 _minContribution
+  ) {
+    hasPaidOut = false;
+    owner = _owner;
+    registry = _registry;
+    donationToken = _donationToken;
+    startTime = _startTime;
+    endTime = _endTime;
+    metaPtr = _metaPtr;
+    minContribution = _minContribution;
+  }
+
+  /**
+   * @notice Before the round ends this method accepts matching pool funds
+   * @param _amount The amount of donation token that can be sent to the contract for the matching pool
+   */
+  function acceptMatchingPool(uint256 _amount) external beforeRoundEnd {
+    _safeTransferDonationToken(msg.sender, address(this), _amount);
```

Comment:
Could we just do `donationToken.safeTransferFrom(msg.sender, address(this), _amount);` instead ?

Reply:
Agree with @thelostone-mc— not sure the overhead of an internal function is worthwhile here. Doesn't seem any more clear than dot syntax.